### DISCUSSION TITLE
Is this the correct behavior for exectution_timeout for a task?

### DISCUSSION THREAD

**Discussion by RafayAK** [apache/airflow#14894] [Q&A]
```bash 
raise AirflowTaskTimeout(self.error_message)
airflow.exceptions.AirflowTaskTimeout: Timeout, PID: 81
[2021-03-19 08:25:00,403] {taskinstance.py:1503} INFO - Marking task as UP_FOR_RETRY. dag_id=backup_dag, task_id=push_targets, execution_date=20210318T020000, start_date=20210319T022500, end_date=20210319T082500
```
I want a task in a dag to run till completion or timeout and fail, right now AIrflow has instead set it up for a retry after a timeout. 

`execution_timeout=timedelta(hours=6)` and `'retries': 2`, retries have been set for transient errors and exection_timeout for the entire task

**Comment by turbaszek** [comment]
Have you tried setting `retries=0` in your task or changing `default_task_retries` in config? You can also set it on dag level.

**Reply by RafayAK** [reply]
@turbaszek Yep, that was the problem. As each task runs once by default and further has `retries=2`, it runs a total of `3*6=18 hours`, if the task does not complete during the allocated 6 hours.

Is there a way to fail a task if it times out, instead of setting `retries=0`? I want to keep retires to overcome transient errors but fail without retrying if it reaches its allocated max time.
